commit 9b06d5b956131bde535f5c045cf8c1ff6bfba76c
Author: Jan Stancek <jstancek@redhat.com>
Date:   Thu Jan 23 13:38:51 2025 +0100

    selftests: net/{lib,openvswitch}: extend CFLAGS to keep options from environment
    
    Package build environments like Fedora rpmbuild introduced hardening
    options (e.g. -pie -Wl,-z,now) by passing a -spec option to CFLAGS
    and LDFLAGS.
    
    Some Makefiles currently override CFLAGS but not LDFLAGS, which leads
    to a mismatch and build failure, for example:
      /usr/bin/ld: /tmp/ccd2apay.o: relocation R_X86_64_32 against
        `.rodata.str1.1' can not be used when making a PIE object; recompile with -fPIE
      /usr/bin/ld: failed to set dynamic section sizes: bad value
      collect2: error: ld returned 1 exit status
      make[1]: *** [../../lib.mk:222: tools/testing/selftests/net/lib/csum] Error 1
    
    openvswitch/Makefile CFLAGS currently do not appear to be used, but
    fix it anyway for the case when new tests are introduced in future.
    
    Fixes: 1d0dc857b5d8 ("selftests: drv-net: add checksum tests")
    Signed-off-by: Jan Stancek <jstancek@redhat.com>
    Acked-by: Matthieu Baerts (NGI0) <matttbe@kernel.org>
    Reviewed-by: Hangbin Liu <liuhangbin@gmail.com>
    Link: https://patch.msgid.link/3d173603ee258f419d0403363765c9f9494ff79a.1737635092.git.jstancek@redhat.com
    Signed-off-by: Jakub Kicinski <kuba@kernel.org>

diff --git a/tools/testing/selftests/net/openvswitch/Makefile b/tools/testing/selftests/net/openvswitch/Makefile
index 2f1508abc826..3fd1da2ec07d 100644
--- a/tools/testing/selftests/net/openvswitch/Makefile
+++ b/tools/testing/selftests/net/openvswitch/Makefile
@@ -2,7 +2,7 @@
 
 top_srcdir = ../../../../..
 
-CFLAGS =  -Wall -Wl,--no-as-needed -O2 -g -I$(top_srcdir)/usr/include $(KHDR_INCLUDES)
+CFLAGS += -Wall -Wl,--no-as-needed -O2 -g -I$(top_srcdir)/usr/include $(KHDR_INCLUDES)
 
 TEST_PROGS := openvswitch.sh